lib/bumblebee/logits_processor.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Bumblebee.LogitsProcessor do
+  @moduledoc """
+  An interface for configuring and using logits processors.
+
+  Logits processors are used during autoregressive generation to modify
+  predicted scores at each generation step. This allows for applying
+  certain rules to the model output to control which tokens are picked
+  at each generation step, and which are not.
+
+  Every module implementing this behaviour is expected to also define
+  a configuration struct.
+  """
+
+  @type t :: Bumblebee.Configurable.t()
+
+  @type state :: Nx.Container.t()
+
+  @type process_context :: %{
+    sequence: Nx.Tensor.t(),
+    length: Nx.Tensor.t(),
+    input_length: Nx.Tensor.t()
+  }
```

Comment:
CI fails formatting:

```suggestion
  @type process_context :: %{
          sequence: Nx.Tensor.t(),
          length: Nx.Tensor.t(),
          input_length: Nx.Tensor.t()
        }
```